src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java

```diff
@@ -0,0 +1,144 @@
+ * @see com.powsybl.commons.reporter.ReporterModel
+ * @see com.powsybl.commons.reporter.Report
+ */
+@SuppressWarnings("SwitchStatementWithTooFewBranches") //for future additions
+@Slf4j
+@Component
+public class ReportMapper {
+    /**
+     * Will try to modify the reporter
+     * @param reporter the reporter to modify
+     * @return the result
+     *
+     * @apiNote is limited to some implementations of {@link Reporter}
+     */
+    public Reporter modifyReporter(@NonNull final Reporter reporter) {
+        if (reporter instanceof ReporterModel reporterModel) {
+            log.trace("ReportModel found");
+            return modifyReporterModel(reporterModel);
+        } else {
+            log.trace("Unrecognized Reporter of type {}", reporter.getClass().getSimpleName());
+            return reporter;
+        }
+    }
+
+    /**
+     * Starting point, determine the root type
+     */
+    protected Reporter modifyReporterModel(@NonNull final ReporterModel reporterModel) {
+        if (reporterModel.getTaskKey().matches("^\\d{8}-\\d{4}-\\d{4}-\\d{4}-\\d{12}@ShortCircuitAnalysis$")) {
+            log.debug("ShortCircuitAnalysis root node found, will modify it!");
+            return forUuidAtShortCircuitAnalysis(reporterModel);
+        } else {
+            log.debug("No treatment for root node {}", reporterModel.getTaskKey());
+            return reporterModel;
+        }
+    }
+
+    /**
+     * Modify node with key {@code ********-****-****-****-************@ShortCircuitAnalysis}
+     *
+     * @implNote we assume there will always be at least one modification
+     */
+    protected Reporter forUuidAtShortCircuitAnalysis(@NonNull final ReporterModel reporterModel) {
+        final ReporterModel newReporter = new ReporterModel(reporterModel.getTaskKey(), reporterModel.getDefaultName(), reporterModel.getTaskValues());
+        reporterModel.getReports().forEach(newReporter::report);
+        reporterModel.getSubReporters().forEach(reporter -> newReporter.addSubReporter(
+            switch (reporter.getTaskKey()) {
+                case "ShortCircuitAnalysis" -> forShortCircuitAnalysis(reporter);
+                default -> reporter;
+            }));
+        return newReporter;
+    }
+
+    /**
+     * Modify node with key {@code ShortCircuitAnalysis}
+     *
+     * @implNote we assume there will always be at least one modification
+     */
+    protected ReporterModel forShortCircuitAnalysis(@NonNull final ReporterModel reporterModel) {
+        final ReporterModel newReporter = new ReporterModel(reporterModel.getTaskKey(), reporterModel.getDefaultName(), reporterModel.getTaskValues());
+        reporterModel.getReports().forEach(newReporter::report);
+        reporterModel.getSubReporters().forEach(reporter -> newReporter.addSubReporter(
+            switch (reporter.getTaskKey()) {
+                case "generatorConversion" -> forGeneratorConversion(reporter);
+                case "courcirc" -> forCourcirc(reporter);
```

Comment:
@geofjamg is this acceptable ? same for "forCourcirc" method ?

Reply:
I have else separate with an interface in [another branch](/gridsuite/shortcircuit-server/tree/feature/concat_courcic_logs), so we can keep the ShortCircuitReports class in our package.